src/common/config.cpp

```diff
@@ -128,13 +129,21 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
     }
 
     const auto conf_path{GetConfigFilePath()};
-    std::ifstream stream{conf_path};
+    std::ifstream stream;
+    if (!conf_path.empty()) { // path is empty when -noconf is specified
+        if (fs::is_directory(conf_path)) {
+            error = strprintf("config file \"%s\" is a directory.", fs::PathToString(conf_path));
+            return false;
+        }
+        stream = std::ifstream{conf_path};
 
-    // not ok to have a config file specified that cannot be opened
-    if (IsArgSet("-conf") && !stream.good()) {
-        error = strprintf("specified config file \"%s\" could not be opened.", fs::PathToString(conf_path));
-        return false;
+        // not ok to have a config file specified that cannot be opened
```

Comment:
nit: double negation makes it more complicated than need be. If you insist on having a comment:
```suggestion
        // Config file must be accessible
```

Reply:
Good point about double negation. Changed to:
```C++
// If the file is explicitly specified, it must be readable
```
since the distinction is important.